src/20241104_AaveV3EthereumLido_OnboardGHOAndMigrateStreamsToLidoInstance/AaveV3EthereumLido_OnboardGHOAndMigrateStreamsToLidoInstance_20241104.sol

```diff
@@ -0,0 +1,209 @@
+      100
+    );
+
+    // usdc
+    AaveV3Ethereum.COLLECTOR.withdrawFromV3(
+      CollectorUtils.IOInput({
+        pool: address(AaveV3Ethereum.POOL),
+        underlying: AaveV3EthereumAssets.USDC_UNDERLYING,
+        amount: A_ETH_USDC_WITHDRAW_AMOUNT
+      }),
+      MiscEthereum.AAVE_SWAPPER
+    );
+
+    AaveSwapper(MiscEthereum.AAVE_SWAPPER).swap(
+      MILKMAN,
+      PRICE_CHECKER,
+      AaveV3EthereumAssets.USDC_UNDERLYING,
+      AaveV3EthereumAssets.GHO_UNDERLYING,
+      AaveV3EthereumAssets.USDC_ORACLE,
+      GHO_USD_FEED,
+      SELF,
+      IERC20(AaveV3EthereumAssets.USDC_UNDERLYING).balanceOf(MiscEthereum.AAVE_SWAPPER),
+      100
+    );
+  }
+
+  function newListings() public pure override returns (IAaveV3ConfigEngine.Listing[] memory) {
+    IAaveV3ConfigEngine.Listing[] memory listings = new IAaveV3ConfigEngine.Listing[](1);
+
+    listings[0] = IAaveV3ConfigEngine.Listing({
+      asset: AaveV3EthereumAssets.GHO_UNDERLYING,
+      assetSymbol: 'GHO',
+      priceFeed: AaveV3EthereumAssets.GHO_ORACLE,
+      enabledToBorrow: EngineFlags.ENABLED,
+      borrowableInIsolation: EngineFlags.DISABLED,
+      withSiloedBorrowing: EngineFlags.DISABLED,
+      flashloanable: EngineFlags.ENABLED,
+      ltv: 0,
+      liqThreshold: 0,
+      liqBonus: 0,
+      reserveFactor: 10_00,
+      supplyCap: 20_000_000,
+      borrowCap: 2_500_000,
+      debtCeiling: 0,
+      liqProtocolFee: 0,
+      rateStrategyParams: IAaveV3ConfigEngine.InterestRateInputData({
+        optimalUsageRatio: 92_00,
+        baseVariableBorrowRate: 10_50,
+        variableRateSlope1: 3_00,
+        variableRateSlope2: 50_00
+      })
+    });
+
+    return listings;
+  }
+
+  function assetsEModeUpdates()
+    public
+    pure
+    override
+    returns (IAaveV3ConfigEngine.AssetEModeUpdate[] memory)
+  {
+    IAaveV3ConfigEngine.AssetEModeUpdate[]
+      memory assetEModeUpdates = new IAaveV3ConfigEngine.AssetEModeUpdate[](3);
+
+    assetEModeUpdates[0] = IAaveV3ConfigEngine.AssetEModeUpdate({
+      asset: AaveV3EthereumAssets.GHO_UNDERLYING,
+      eModeCategory: AaveV3EthereumLidoEModes.SUSDE_STABLECOINS,
+      borrowable: EngineFlags.ENABLED,
+      collateral: EngineFlags.DISABLED
+    });
+
+    assetEModeUpdates[1] = IAaveV3ConfigEngine.AssetEModeUpdate({
+      asset: AaveV3EthereumAssets.USDC_UNDERLYING,
+      eModeCategory: AaveV3EthereumLidoEModes.LRT_STABLECOINS_MAIN,
+      borrowable: EngineFlags.ENABLED,
+      collateral: EngineFlags.DISABLED
+    });
+
+    assetEModeUpdates[2] = IAaveV3ConfigEngine.AssetEModeUpdate({
+      asset: AaveV3EthereumAssets.GHO_UNDERLYING,
+      eModeCategory: AaveV3EthereumLidoEModes.LRT_STABLECOINS_MAIN,
+      borrowable: EngineFlags.ENABLED,
+      collateral: EngineFlags.DISABLED
+    });
+
+    return assetEModeUpdates;
+  }
+
+  /**
+   * @notice deposit token to AaveV3EthereumLido pool
+   * @param token The address of token to deposit
+   */
+  function deposit(address token) external {
```

Comment:
@sendra maybe let's add a rescue function callable by a guardian to rescue erc20 tokens just in case?
we've used this pattern before but still, i always like to be safe.

Reply:
makes sense, lets add